examples/chat_app/interface/hddr_llm_onnx_dml_interface.py

```diff
@@ -0,0 +1,182 @@
+import gc
+import os
+import sys
+import onnxruntime_genai as og
+from app_modules.utils import convert_to_markdown, is_stop_word_or_prefix, shared_state
+
+current_dir = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(current_dir, "..", "..", ".."))
+
+
+class ONNXModel():
+    """A wrapper for OnnxRuntime-GenAI to run ONNX LLM model."""
+
+    def __init__(self, model_path):
+        self.og = og
+        self.model = og.Model(f'{model_path}')
+        self.tokenizer = og.Tokenizer(self.model)
+        self.tokenizer_stream = self.tokenizer.create_stream()
+        self.is_phi = "phi" in model_path
+        if self.is_phi:
+            self.enable_history_max = 10 if "mini" in model_path else 2
+            self.history_template = "<|user|>{input}<|end|><|assistant|>{response}<|end|>"
+            self.chat_template = "<|user|>{input}<|end|><|assistant|>"
+        else:
+            self.enable_history_max = 2
+            self.history_template = "[INST]{input}[/INST]{response}"
+            self.chat_template = "[INST]{input}[/INST]"
+
+    def generate_prompt_with_history(self, text, history, max_length=2048):
+        prompt = ""
+
+        for dialog in history[-self.enable_history_max:]:
+            prompt += f'{self.history_template.format(input=dialog[0], response=dialog[1])}'
+
+        if not self.is_phi:
+            prompt = f"<s>{prompt}</s>"
+
+        prompt += f'{self.chat_template.format(input=text)}'
+
+        input_ids = self.tokenizer.encode(prompt + f'{self.chat_template.format(input=text)}')
+
+        if len(input_ids) <= max_length:
+            return input_ids
+        else:
+            history.clear()
+            return self.tokenizer.encode(f'{self.chat_template.format(input=text)}')
+
+    def greedy_search(
```

Comment:
why fix it to greedy search?

Reply:
Changed the name to search and default search strategy is applied